tardis/energy_input/tests/test_gamma_ray_transport.py

```diff
@@ -0,0 +1,195 @@
+import os
+import pytest
+import numpy as np
+import numpy.testing as npt
+from pathlib import Path
+import radioactivedecay as rd
+
+import tardis
+from tardis.model import SimulationState
+from tardis.io.configuration import config_reader
+from tardis.io.util import yaml_load_file, YAMLLoader
+from tardis.energy_input.gamma_ray_transport import (
+    calculate_shell_masses,
+    create_isotope_dicts,
+    get_all_isotopes,
+    create_inventories_dict,
+    calculate_total_decays,
+)
+import astropy.units as u
+import astropy.constants as c
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_config(example_configuration_dir: Path):
+    """
+    Parameters
+    ----------
+    example_configuration_dir: Path to the configuration directory.
+
+    Returns
+    -------
+    Tardis configuration
+    """
+    yml_path = (
+        example_configuration_dir
+        / "tardis_configv1_density_exponential_nebular.yml"
+    )
+
+    return config_reader.Configuration.from_yaml(yml_path)
+
+
+@pytest.fixture(scope="module")
+def simulation_setup(gamma_ray_config):
+    """
+    Parameters
+    ----------
+    gamma_ray_config:
+
+    Returns
+    -------
+
+    """
+    gamma_ray_config.model.structure.velocity.start = 1.0 * u.km / u.s
+    gamma_ray_config.model.structure.density.rho_0 = 5.0e2 * u.g / (u.cm**3)
+    gamma_ray_config.supernova.time_explosion = 1.0 * (u.d)
+    model = SimulationState.from_config(gamma_ray_config)
+    return model
+
+
+def test_calculate_shell_masses(simulation_setup):
```

Comment:
I'm not sure this is a great test. If I understand this, this test is effectively doing the exact same calculation as the calculate_shell_masses() function, but just taking .value after, so if something changes in that function we wouldn't particularly see a difference. I think this is a place I would use parametrize and just pick a couple values that you calculate by hand for desired, and just check those against the values output by calculate_shell_masses().

Reply:
thankyou @jvshields. I did this to test if any restructure affects the model. Maybe I have to find a better way to do this.

Reply:
Added hand calculated values.